#! /bin/sh /usr/share/dpatch/dpatch-run
## 12_fix_screensaver_in_mythgame.dpatch by Michael Haas  <laga@laga.ath.cx>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Disable deinterlacer when starting a game

@DPATCH@
diff -urNad mythplugins-fixes~/mythgame/mythgame/gamehandler.cpp mythplugins-fixes/mythgame/mythgame/gamehandler.cpp
--- mythplugins-fixes~/mythgame/mythgame/gamehandler.cpp	2008-09-26 20:51:42.000000000 +0200
+++ mythplugins-fixes/mythgame/mythgame/gamehandler.cpp	2008-09-26 20:53:22.000000000 +0200
@@ -858,6 +858,7 @@
 		              .arg(handler->SystemName())
 			      .arg(exec));
 
+    gContext->DoDisableScreensaver();
 
     QStringList cmdlist = QStringList::split(";", exec);
     if (cmdlist.count() > 0) {
@@ -871,7 +872,9 @@
         myth_system(exec);
     }
 
-    (void)d.cd(savedir);      
+    gContext->DoRestoreScreensaver();
+
+    (void)d.cd(savedir);        
 }
 
 RomInfo* GameHandler::CreateRomInfo(RomInfo* parent)
